persistence/persistence-mybatis-test/src/test/java/org/cbioportal/persistence/mybatis/CancerTypeMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,111 @@
+package org.cbioportal.persistence.mybatis;
+
+import org.cbioportal.model.TypeOfCancer;
+import org.cbioportal.model.meta.BaseMeta;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration("/testContextDatabase.xml")
+@Configurable
+public class CancerTypeMyBatisRepositoryTest {
+
+    @Autowired
+    private CancerTypeMyBatisRepository cancerTypeMyBatisRepository;
+
+    @Test
+    public void getAllCancerTypesIdProjection() throws Exception {
+
+        List<TypeOfCancer> result = cancerTypeMyBatisRepository.getAllCancerTypes("ID", null, null, null, null);
+
+        Assert.assertEquals(2, result.size());
+        TypeOfCancer typeOfCancer = result.get(0);
```

Comment:
Since we are using an in-memory H2 database for these tests, not a 3rd party MySql that can have different versions or different storage engines, the tests should be deterministic. So yes you can't guarantee a particular order, it's up to the database implementation, but you won't get a random order every time you call it, it will be the same every time if the inserted data is the same. And when we change the inserted data, we need to change some tests in any case.